File: 75_sort_colors/sortColors.java
class Solution {
    public void sortColors(int[] nums) {
        int[] count = new int[4];

        for (int num : nums) {
            count[num]++;
        }

        int pos = 0;

        for (int i = 0; i < count.length; i++) {
            while (count[i] > 0) {
                nums[pos] = i;
                count[i]--;
                pos++;
            }
        }
    }
}

File: 75_sort_colors/sortColors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int count[3] = {0, 0, 0};
        int n = nums.size();

        for (int num : nums) {
            count[num]++;
        }

        int k = 0;

        for (int i = 0; i < sizeof(count)/sizeof(*count); i++) {
            for (int j = 0; j < count[i]; j++) {
                nums[k] = i;
                k++;
            }
        }

        return;
    }
};